spinel/codec.py

```diff
@@ -119,13 +119,11 @@ def parse_e(cls, payload):
 
     @classmethod
     def parse_U(cls, payload):
-        payload = payload.decode('utf-8')
-        nullchar = '\0'
-
+        nullchar = b'\0'
         if payload.find(nullchar) >= 0:
-            return payload[:payload.index(nullchar)]  # strip null
+            return payload[:payload.index(nullchar)].decode('utf-8')  # strip null
```

Comment:
```suggestion
            return payload[:payload.index(nullchar)].decode('utf-8')  # strip null
```
nit: no need to call `index` again since we already have the index by `.find`?